pkg/server/commands/listusers/validate.go

```diff
@@ -0,0 +1,95 @@
+package listusers
+
+import (
+	"context"
+	"errors"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+
+	serverErrors "github.com/openfga/openfga/pkg/server/errors"
+
+	"github.com/openfga/openfga/internal/validation"
+	"github.com/openfga/openfga/pkg/typesystem"
+)
+
+func ValidateListUsersRequest(ctx context.Context, req *openfgav1.ListUsersRequest, typesys *typesystem.TypeSystem) error {
+	_, span := tracer.Start(ctx, "validateListUsersRequest")
+	defer span.End()
+	if err := validateContextualTuples(req, typesys); err != nil {
+		return err
+	}
+
+	if err := validateUsersFilters(req, typesys); err != nil {
+		return err
+	}
+
+	return validateTargetRelation(req, typesys)
+}
+
+func validateContextualTuples(request *openfgav1.ListUsersRequest, typeSystem *typesystem.TypeSystem) error {
+	for _, contextualTuple := range request.GetContextualTuples() {
```

Comment:
We now have this duplicated at least 3 times (Check, ListObjects, ListUsers). We should move this into `internal/validation` in the future and consolidate.